sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3046,6 +3046,14 @@ object functions {
     ArrayContains(column.expr, Literal(value))
   }
 
+  /**
+   * Merges multiple arrays into one by putting elements from the specific array after elements
+   * from the previous array. If any of the arrays is null, null is returned.
+   * @group collection_funcs
+   * @since 2.4.0
+   */
+  def concat_arrays(columns: Column*): Column = withExpr { ConcatArrays(columns.map(_.expr)) }
```

Comment:
We need to add this func. in `sql/functions` here? It seems we might recommend users to use these kinds of functions via `selectExpr`, so is it okay to add this only in `FunctionRegistry`? Thoughts? @viirya @gatorsmile